cluster-autoscaler/cloudprovider/aws/aws_cloud_provider.go

```diff
@@ -319,9 +319,9 @@ func (ng *AwsNodeGroup) Belongs(node *apiv1.Node) (bool, error) {
 }
 
 // DeleteNodes deletes the nodes from the group.
-func (ng *AwsNodeGroup) DeleteNodes(nodes []*apiv1.Node) error {
+func (ng *AwsNodeGroup) DeleteNodes(nodes []*apiv1.Node, respectMinCount bool) error {
 	size := ng.asg.curSize
-	if int(size) <= ng.MinSize() {
+	if int(size) <= ng.MinSize() && respectMinCount {
```

Comment:
when `respectMinCount = false` and nodes has placeholder, the `DeleteInstances` will `setAsgSize`.  Will new size < min size will resturn an error?